op-challenger/fault/monitor.go

```diff
@@ -66,18 +66,28 @@ func (m *gameMonitor) progressGames(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("failed to load games: %w", err)
 	}
+	requiredGames := make(map[common.Address]bool)
 	for _, game := range games {
 		if !m.allowedGame(game.Proxy) {
 			m.logger.Debug("Skipping game not on allow list", "game", game.Proxy)
 			continue
 		}
+		requiredGames[game.Proxy] = true
 		player, err := m.fetchOrCreateGamePlayer(game)
 		if err != nil {
 			m.logger.Error("Error while progressing game", "game", game.Proxy, "err", err)
 			continue
 		}
 		player.ProgressGame(ctx)
```

Comment:
I think this currently will only delete games that are not in the `allowedGame` list. Instead, we should be checking the bool return value of `ProgressGame` which will return true if the game is no longer in progress. We can then only add the game proxy address to the `requiredGames` mapping when the `ProgressGame` return value is false.

A way to do this would look be the following.
```suggestion
		if !m.allowedGame(game.Proxy) {
			m.logger.Debug("Skipping game not on allow list", "game", game.Proxy)
			continue
		}
		player, err := m.fetchOrCreateGamePlayer(game)
		if err != nil {
			m.logger.Error("Error while progressing game", "game", game.Proxy, "err", err)
			continue
		}
		requiredGames[game.Proxy] = !player.ProgressGame(ctx)
```

Reply:
If we do it that way the next update will create the player again because it's still in the list of games to progress. Since creating the player involves a bunch of calls but it only uses a small amount of memory I think we want to keep the player around in memory for as long as its present in the list of games we care about, rather than drop it from memory when `ProgressGame` returns true.

We can still delete the data from disk when `ProgressGame` returns true, we just need something to remember the details of the game to save us loading them all on every update.

Reply:
Ah got it - I didn't intuit the on disk / in memory resource split here until our sync call. That makes sense to me!